service/handler_secretkey.go

```diff
@@ -0,0 +1,101 @@
+package service
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/gorilla/mux"
+	"github.com/medibloc/panacea-oracle/crypto"
+	"github.com/medibloc/panacea-oracle/server/middleware"
+	log "github.com/sirupsen/logrus"
+)
+
+func (svc *Service) GetSecretKey(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'm worried about a lot of code using queryClient.
This causes handler response to be delayed.

Umm We need to solution this part.

Reply:
Agree. `ValidateData` as well.
As we discussed before, we can cache all the things from verified query client. Let's make improvement further.